### PR TITLE
added Spring Data JPA, liquibase support, soft delete concept and missing endpoints

### PR DESCRIPTION
**Pull request by ArtemReizer**

### CODE REVIEW

**Review by OlegTuryk — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -20,6 +20,10 @@
         <mapstruct.version>1.5.5.Final</mapstruct.version>
 	</properties>
 	<dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
```
Comment (vovasalyha) [inline pom.xml]: do you really need this dependency?

File: src/main/java/com/app/bookstore/repository/book/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.app.bookstore.repository.book;
+
+import com.app.bookstore.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    Book getBookById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/app/bookstore/repository/book/BookRepository.java]: you don't need this. findById which returns optional is the best choice.

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: empty line

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Please take a look on my comments. Also, resolve all merge conflicts before asking a review next time.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**